signal_generator.py

```
@@ -855,6 +855,212 @@ def _combine_signals(self, fib_analysis: Dict, tails_analysis: Dict,
+                        vol_ratio = vol_data.get('volume_ratio', 0.0)
+                        if vol_ratio >= 1.5:  # 1.5x average volume minimum
+                            confluence_count += 1
+                            volume_confirmed = True
+                            confluence_bonus += 0.05
+                            quality_factors.append(f"Volume confirmation ({vol_ratio:.1f}x avg)")
+                        else:
+                            warning_factors.append(f"Low volume ({vol_ratio:.1f}x avg)")
+                    
+                    # Bonus confirmations for additional quality
+                    # Multi-timeframe alignment confirmation
+                    if 'multi_timeframe_analysis' in locals() and multi_timeframe_analysis:
```

Ruff (0.12.2)

Undefined name `multi_timeframe_analysis`

(F821)